### PR TITLE
expand ingest test coverage

### PR DESCRIPTION
**Pull request by fvankrieken**
100% coverage of ingest

Clearly needed and not just out of principle - a few little bugs found (mainly to do with index of pd dataframes)

I'll squash this. Commit-by-commit might be nice, it is organized by type of test (1 commit for `configure`, another for `extract`, many for different parts of `transform`

The final commit makes coverage reports ignore typer CLI commands. If anyone has objections, happy to hear it, but it made sense to ignore them. Calling them via python seems to be a little odd - typer.Option(None, ...) as a default resolves to `None` when you invoke CLI, but if you just try calling the function, an actual `typer` object gets passed around (and odd behavior results)

### CODE REVIEW

File: dcpy/lifecycle/ingest/extract.py [hunk 1]
```diff
@@ -35,11 +36,10 @@ def download_file_from_source(
         case S3Source():
             s3.download_file(source.bucket, source.key, path)
         case ScriptSource():
```
Comment (fvankrieken) [inline dcpy/lifecycle/ingest/extract.py:38]: This still isn't actually used yet at all, so it's a little weird at the moment. Will likely be renamed/refactored at some point (or removed entirely)

File: dcpy/test/lifecycle/ingest/test_transform.py [hunk 2]
```diff
@@ -106,85 +100,212 @@ def test_validate_processing_steps():
...
+        # Invalid pd series func
+        PreprocessingStep(
+            name="pd_series_func",
+            args={"function_name": "str.fake_function", "column_name": "_"},
+        ),
+    ],
+)
+def test_errors(step):
+    with pytest.raises(Exception, match="Invalid preprocessing steps"):
+        transform.validate_processing_steps("test", [step])
```
[the remark this answers is not in the document]
Comment (fvankrieken) [reply dcpy/test/lifecycle/ingest/test_transform.py:125]: What do you mean?

File: dcpy/test/lifecycle/ingest/test_run.py
Comment (damonmcc) [inline dcpy/test/lifecycle/ingest/test_run.py:1]: (just an idea for later)

these test use the `edm-recipes` bucket, but could they use the mocked bucket from `conftest.py` [here](https://github.com/NYCPlanning/data-engineering/blob/main/dcpy/test/conftest.py)?

File: dcpy/test/lifecycle/ingest/test_transform.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fvankrieken) [reply dcpy/test/lifecycle/ingest/test_transform.py:125]: It's a test, though it definitely needs an actually descriptive name

[the remark this answers is not in the document]
Comment (fvankrieken) [reply dcpy/test/lifecycle/ingest/test_transform.py]: Changed in the latest commit

File: dcpy/test/lifecycle/ingest/test_run.py
Comment (fvankrieken) [reply dcpy/test/lifecycle/ingest/test_run.py:1]: Currently there is no override for any calls to `recipes` connector, so in the mocked s3 environment datasets still end up in the recipes bucket

**Comment by fvankrieken** [comment]
Just had a test fail seemingly unrelated to the minor commit I pushed, hope I've sorted it out but might need to add a commit or two

**Comment by fvankrieken** [comment]
> Just had a test fail seemingly unrelated to the minor commit I pushed, hope I've sorted it out but might need to add a commit or two

It was a flaky test I had written, fixed now (forgot it was using the run date as a version)

File: dcpy/test/lifecycle/ingest/test_transform.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (damonmcc) [reply dcpy/test/lifecycle/ingest/test_transform.py]: my bad for some reason I thought this was a fixture

**Review by damonmcc — APPROVED** [review]
love it!

File: dcpy/test/lifecycle/ingest/test_transform.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fvankrieken) [reply dcpy/test/lifecycle/ingest/test_transform.py]: It did have the entirely unhelpful name `test_errors`

**Pull request merged**